Trappist/src/Promact.Trappist.Core/Controllers/CategoryController.cs

```diff
@@ -23,50 +27,54 @@ public IActionResult GetAllCategories()
             return Ok(categoryList);
         }
 
-        #region post Method 
         [HttpPost]
         /// <summary>
-        /// Post Method 
-        /// Will Add a Catagory Name in Category Model
+        /// Method to Add Category
         ///</summary>
-        /// <param name="category">Object of  class Category</param>
-        /// <returns>object of the class </returns>
-        public IActionResult CatagoryAdd([FromBody] Category category)
+        /// <param name="category">category object contains category details</param>
+        /// <returns>category object contains category details</returns>
+        public async Task<IActionResult> AddCategory([FromBody] Category category)
         {
             if (!ModelState.IsValid)
             {
                 return BadRequest(ModelState);
             }
-            _categoryRepository.AddCategory(category);
+            var isCategoryNameExists = await _categoryRepository.CheckDuplicateCategoryNameAsync(category.CategoryName);
+            if (isCategoryNameExists == true)
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Remove the unnecessary Boolean literal(s). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://server12:9005/coding_rules#rule_key=csharpsquid%3AS1125)